src/openrct2/interface/Viewport.cpp

```diff
@@ -836,9 +836,26 @@ void viewport_render(
 #endif
 }
 
-static void viewport_fill_column(paint_session* session)
+static void viewport_fill_column(paint_session* session, std::vector<paint_session>* sessions, size_t index)
 {
     paint_session_generate(session);
+    // Perform a deep copy of the paint session, use relative offsets.
+    // This is done to extract the session for benchmark.
+    if (sessions != nullptr)
+    {
+        (*sessions)[index] = (*session);
```

Comment:
I think it would be better to move the code into a new function to better separate the testing code out visibly.